br/pkg/restore/split/splitter.go

```diff
@@ -0,0 +1,401 @@
+// Copyright 2024 PingCAP, Inc. Licensed under Apache-2.0.
+
+package split
+
+import (
+	"bytes"
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	restoreutils "github.com/pingcap/tidb/br/pkg/restore/utils"
+	"github.com/pingcap/tidb/pkg/tablecodec"
+	"github.com/pingcap/tidb/pkg/util"
+	"github.com/pingcap/tidb/pkg/util/codec"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+// Splitter defines the interface for basic splitting strategies.
+type Splitter interface {
+	// ExecuteSortedKeysOnRegion splits the keys within a single region and initiates scattering
+	// after the region has been successfully split.
+	ExecuteSortedKeysOnRegion(ctx context.Context, region *RegionInfo, keys [][]byte) ([]*RegionInfo, error)
+
+	// ExecuteSortedKeys splits all provided keys while ensuring that the newly created
+	// regions are balanced. It first applies the rewrite rules, then splits regions
```

Comment:
It seems there isn't rewrite rules in the argument... Perhaps move the comments about rewrite rules to the implement.